lesson22/main.js

```diff
@@ -145,56 +145,85 @@ const createSortBtn = () => {
   return btn;
 };
 
-const renderSortBtn = () => {
-  const btn = createSortBtn();
-  const targetEle = document.getElementById("id");
-  targetEle.appendChild(btn);
+const setDataSort = (usersData) => {
+  Object.keys(usersData[0]).forEach((key) => {
+    if (targetSortElements.includes(key)) {
+      const element = document.getElementById(key);
+      element.dataset.sort = "sort";
+    }
+  });
+};
+
+const renderSortBtn = (usersData) => {
+  setDataSort(usersData);
+  const elements = document.querySelectorAll("[data-sort]");
+  Array.from(elements).forEach((element) => {
+    const btn = createSortBtn();
+    element.appendChild(btn);
+  });
 };
 
 const clickSortBtn = (usersData) => {
-  const sortArrow = document.querySelector(".sort-btn");
-  const sortArrowImg = document.querySelector(".sort-img");
-  sortArrow.addEventListener("click", () => {
-    changeSortStateAndArrowImg(sortArrowImg);
-    rerenderTableData(usersData);
+  const sortArrowBtns = document.querySelectorAll(".sort-btn");
+  Array.from(sortArrowBtns).forEach((sortArrowBtn) => {
+    const tHeaderId = sortArrowBtn.parentNode.id;
+    const sortArrowImg = sortArrowBtn.firstElementChild;
+    sortArrowBtn.addEventListener("click", () => {
+      initSortAndOthersSortImg(tHeaderId);
+      changeSortStateAndArrowImg(sortArrowImg);
+      rerenderTableData(usersData, tHeaderId);
+    });
   });
 };
 
-const changeSortStateAndArrowImg = (sortArrow) => {
+const initSortAndOthersSortImg = (tHeaderId) => {
+  const othersId = targetSortElements.find((name) => name !== tHeaderId);
+  const othersBtn = document.getElementById(othersId).firstElementChild;
+  const othersImg = othersBtn.firstElementChild;
+  if (othersImg.getAttribute("src") !== "/img/both.svg") {
+    othersImg.src = "/img/both.svg";
+    sortState = "both";
+  }
+};
+
+const changeSortStateAndArrowImg = (sortArrowImg) => {
   if (sortState === "both") {
     sortState = "asc";
-    sortArrow.src = "/img/asc.svg";
+    sortArrowImg.src = "/img/asc.svg";
     return;
   }
   if (sortState === "asc") {
     sortState = "desc";
-    sortArrow.src = "/img/desc.svg";
+    sortArrowImg.src = "/img/desc.svg";
     return;
   }
   sortState = "both";
-  sortArrow.src = "/img/both.svg";
+  sortArrowImg.src = "/img/both.svg";
```

Comment:
IMO
i often use switch because of readable

Reply:
I think it's much better now! Thank you.
[4a0fd5a](https://github.com/yuka830/js-lessons/pull/28/commits/4a0fd5a13aab7e042ccbee72a96e4126178d0205)